h/services/user_signup.py

```diff
@@ -80,7 +86,23 @@ def signup(self, require_activation=True, **kwargs):
 
         # Create a new activation for the user
         if require_activation:
-            self._require_activation(user)
+            try:
+                self._require_activation(user)
+            except IntegrityError as e:
+                if (
+                    'duplicate key value violates unique constraint "uq__user__email"'
+                    in e.args[0]
+                    or 'duplicate key value violates unique constraint "ix__user__userid"'
+                    in e.args[0]
+                ):
+                    log.warning("email conflict error occured during user signup")
+                    raise ConflictError(
+                        "The email account {} has already been registered.".format(
+                            user.email
```

Comment:
Since we are checking for two different conflicts but always telling the user that only one of them occurred, I think we either need to:

- Add a comment, like you did in the tests, to indicate why we are doing this OR
- Change the message to reflect the two possible causes of the error OR
- Report a different message depending on the exception